src/machine/memory/tlb/tlb.cpp

```diff
@@ -0,0 +1,274 @@
+#include "tlb.h"
+
+#include "csr/controlstate.h"
+#include "machine.h"
+#include "memory/virtual/page_table_walker.h"
+#include "memory/virtual/sv32.h"
+
+LOG_CATEGORY("machine.TLB");
+
+namespace machine {
+
+static bool is_mmio_region(uint64_t virt) {
```

Comment:
This function should be removed from the history. The logic is corrected by the commit https://github.com/cvut/qtrvsim/pull/165/commits/667bd4fa304bc2e71e31e7c5dc3d668d4ca43d21
But it would be much better, if it does not appear in the history at all.